nemo/collections/nlp/models/language_modeling/megatron_t5_speechlm_model.py

```diff
@@ -576,25 +597,29 @@ def validation_step(self, batch, batch_idx, inference=False):
         first_layer_loss = torch.sum(first_layer_loss * loss_mask) / total_predictions
 
         metrics = {
-            'loss': loss_mean,
+            'loss': loss_mean * 0.14 if torch.count_nonzero(speech_mask) > 0 else loss_mean,
             'first_layer_accuracy': first_layer_accuracy,
             'first_layer_loss': first_layer_loss,
         }
         loss_total = first_layer_loss
         for i in range(7):
-            speech_logits_i = speech_logits[:, :, :, i]
-            speech_preds_i = speech_logits_i.argmax(dim=2)  # (t,bs)
-            speech_preds_i = speech_preds_i.transpose(0, 1)  # (bs,t)
-            labels_i = labels_original[:, i + 1, :]  # (bs,t)
-            correct_predictions_i = speech_preds_i == labels_i  # (bs,t)
-            correct_predictions_i = correct_predictions_i * loss_mask * speech_mask  # (bs,t)
-            total_correct_predictions_i = torch.sum(correct_predictions_i)
-            total_predictions_i = torch.sum(loss_mask * speech_mask)
-            speech_accuracy_i = total_correct_predictions_i / total_predictions_i
-            loss_i = torch.nn.functional.cross_entropy(
-                speech_logits_i.permute(1, 2, 0), labels_i, reduction='none'
-            )  # (bs,t)
-            loss_i = torch.sum(loss_i * loss_mask * speech_mask) / total_predictions_i
+            if torch.count_nonzero(speech_mask) > 0:
```

Comment:
Same comment here. Let's try to log validation metrics for both ASR and TTS using speech_mask.

Reply:
Same reply as above.